crates/ironrdp-pdu/src/utils.rs

```diff
@@ -115,31 +115,38 @@ pub fn write_string_to_cursor(
     character_set: CharacterSet,
     write_null_terminator: bool,
 ) -> PduResult<()> {
-    match character_set {
+    let (buffer, ctx) = match character_set {
         CharacterSet::Unicode => {
             let mut buffer = to_utf16_bytes(value);
             if write_null_terminator {
                 buffer.push(0);
                 buffer.push(0);
             }
-
-            ensure_size!(ctx: "Encode sting (UTF-16)", in: cursor, size: buffer.len());
-            cursor.write_slice(&buffer);
+            (buffer, "Encode sting (UTF-16)")
         }
         CharacterSet::Ansi => {
             let mut buffer = value.as_bytes().to_vec();
             if write_null_terminator {
                 buffer.push(0);
             }
-
-            ensure_size!(ctx: "Encode sting (UTF-8)", in: cursor, size: buffer.len());
-            cursor.write_slice(&buffer);
+            (buffer, "Encode sting (UTF-16)")
```

Comment:
```suggestion
            (buffer, "Encode string (UTF-8)")
```
Also, "Ansi" encoding is different from UTF-8, but IIRC UTF-8 is actually used by freerdp in those cases